### PR TITLE
Create the device tree module

### PR DESCRIPTION
**Pull request by poncovka**
Create the device tree module and add DBus support for viewing and handling devices.

### CODE REVIEW

File: pyanaconda/image.py [hunk 1]
```diff
@@ -172,14 +172,27 @@ def mountImage(isodir, tree):
...
 
-    # Search for devices identified as cdrom along with any other
-    # device that has an iso9660 filesystem. This will catch USB media
-    # created from ISO images.
+    FIXME: This is a temporary workaround for UI.
+    """
+    return find_optical_install_media(devicetree)
+
+
+def find_optical_install_media(devicetree):
```
Comment (jkonecny12) [inline pyanaconda/image.py]: This should be moved to payload module later. Storage module should onlyprovide list of valid devices and the check if it is an installation media should be done in the Payload module instead.

File: pyanaconda/modules/storage/devicetree/handler.py [hunk 2]
```diff
@@ -37,3 +38,47 @@ def storage(self):
...
+    def teardown_device(self, device_name):
+        """Close, or tear down, a device.
+
+        :param device_name: a name of the device
+        """
+        device = self._get_device(device_name)
+        device.teardown(recursive=True)
+
+    def mount_device(self, device_name, mount_point):
+        """Activate the formatting and mount the file system.
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/storage/devicetree/handler.py]: Fixed.

File: pyanaconda/modules/storage/devicetree/handler.py [hunk 3]
```diff
@@ -37,3 +38,47 @@ def storage(self):
...
+
+        :param device_name: a name of the device
+        :param mount_point: a path to the mount point
+        """
+        device = self._get_device(device_name)
+        device.format.setup(mountpoint=mount_point)
+
+    def unmount_device(self, device_name, mount_point):
+        """Deactivate the formatting and unmount the file system.
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/storage/devicetree/handler.py]: Fixed.

File: pyanaconda/image.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/image.py]: Fixed.

**Comment by poncovka** [comment]
Updated.

File: pyanaconda/modules/storage/devicetree/handler_interface.py [hunk 4]
```diff
@@ -28,3 +28,33 @@
...
+    def MountDevice(self, device_name: Str, mount_point: Str):
+        """Activate the formatting and mount the file system.
+
+        :param device_name: a name of the device
+        :param mount_point: a path to the mount point
+        """
+        self.implementation.mount_device(device_name, mount_point)
+
+    def UnmountDevice(self, device_name: Str, mount_point: Str):
+        """Deactivate the formatting and unmount the file system.
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/devicetree/handler_interface.py]: Please remove the ``Activate/Deactive the formatting...`` here too.

Comment (jkonecny12) [reply pyanaconda/modules/storage/devicetree/handler_interface.py]: The same works for `MountDevice`.

Comment (poncovka) [reply pyanaconda/modules/storage/devicetree/handler_interface.py]: Fixed.

**Pull request merged**